### PR TITLE
[ContactStructuralMechanicsApplication] Improve contact strategy

### PR DESCRIPTION
**Pull request by loumalouomega**
* Removing check for zero components of the matrix, and ensuring that the matrix is structurally symmetric
* Improving the simplified semi-smooth method

### CODE REVIEW

**Comment by loumalouomega** [comment]
Ping @RiccardoRossi

**Comment by loumalouomega** [comment]
Ping

**Comment by frastellini** [comment]
It looks fine to me.

**Comment by loumalouomega** [comment]
@frastellini can you approve?

File: applications/ContactStructuralMechanicsApplication/python_scripts/contact_structural_mechanics_implicit_dynamic_solver.py [hunk 1]
```diff
@@ -198,7 +198,7 @@ def _create_linear_solver(self):
         mortar_type = self.contact_settings["mortar_type"].GetString()
         if (mortar_type == "ALMContactFrictional" or mortar_type == "ALMContactFrictionlessComponents"):
             if (self.contact_settings["use_mixed_ulm_solver"].GetBool() == True):
-                self.print_on_rank_zero("::[Contact Mechanical Implicit Dynamic Solver]:: ", "Using MixedULMLinearSolver, definition of ALM parameters recommended")
+                self.print_on_rank_zero("::[Contact Mechanical Static Solver]:: ", "Using MixedULMLinearSolver, definition of ALM parameters recommended")
```
Comment (philbucher) [inline applications/ContactStructuralMechanicsApplication/python_scripts/contact_structural_mechanics_implicit_dynamic_solver.py:201]: this again? :)

**Review by philbucher — APPROVED** [review]
ok

File: applications/ContactStructuralMechanicsApplication/python_scripts/contact_structural_mechanics_implicit_dynamic_solver.py (hunk 1 above)
Comment (loumalouomega) [reply applications/ContactStructuralMechanicsApplication/python_scripts/contact_structural_mechanics_implicit_dynamic_solver.py:201]: Damm

**Pull request merged**